### ISSUE TITLE
Cannot send requests via secure connection

### ISSUE THREAD

**Issue by aaronRaines** [zalopay-oss/jmeter-grpc-request#66]
Hi, 

I am trying to use this plugin to send a GRPC request via Jmeter. I have imported the trusted keys into my Jmeter key store and can see them loading in the debug output - however whenever I try and run the request I see the following error.

`DEBUG i.n.h.s.ReferenceCountedOpenSslEngine: SSL_read failed with 1: OpenSSL error: 268436496 error:10000410:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE`

The same GRPC request works in a standalone application using the same keys used to send single requests. I only seem to have this error while sending requests from this plug in. 

Do you have any suggestions on the cause of this issue ? Do I need to import certificates a different way compared to normal Jmeter certificate locations ?

**Comment by Andrey-Demidovich** [comment]
Hi,

Found almost the same issue. Requests from jmeter-grpc-request plugin fails with this in debug logs:

`2022-06-08 15:17:07,969 DEBUG v.z.b.c.g.ChannelFactory: Create SslContext with NPN_AND_ALPN
2022-06-08 15:17:07,974 DEBUG o.a.j.t.p.AbstractProperty: Running version, executing function
2022-06-08 15:17:08,151 DEBUG i.n.h.s.u.InsecureTrustManagerFactory: Accepting a server certificate: CN=server.address
2022-06-08 15:17:08,152 DEBUG i.n.h.s.SslHandler: [id: 0x3d46fc58, L:/192.168.0.224:49389 - R:mrc.server.address/1.1.1.1:443] HANDSHAKEN: protocol:TLSv1.3 cipher suite:TLS_AES_128_GCM_SHA256
2022-06-08 15:17:08,152 DEBUG i.g.n.NettyClientHandler: [id: 0x3d46fc58, L:/192.168.0.224:49389 - R:mrc.server.address/1.1.1.1:443] OUTBOUND SETTINGS: ack=false settings={ENABLE_PUSH=0, MAX_CONCURRENT_STREAMS=0, INITIAL_WINDOW_SIZE=1048576, MAX_HEADER_LIST_SIZE=8192}
2022-06-08 15:17:08,152 DEBUG i.g.n.NettyClientHandler: [id: 0x3d46fc58, L:/192.168.0.224:49389 - R:mrc.server.address/1.1.1.1:443] OUTBOUND WINDOW_UPDATE: streamId=0 windowSizeIncrement=983041
2022-06-08 15:17:08,173 DEBUG i.n.h.s.ReferenceCountedOpenSslEngine: SSL_read failed with 1: OpenSSL error: 268436572 error:1000045c:SSL routines:OPENSSL_internal:TLSV1_ALERT_CERTIFICATE_REQUIRED
2022-06-08 15:17:08,175 DEBUG i.g.n.NettyClientHandler: [id: 0x3d46fc58, L:/192.168.0.224:49389 ! R:mrc.server.address/1.1.1.1:443] OUTBOUND GO_AWAY: lastStreamId=0 errorCode=2 length=105 bytes=6a617661782e6e65742e73736c2e53534c457863657074696f6e3a206572726f723a31303030303435633a53534c20726f7574696e65733a4f50454e53534c5f...
2022-06-08 15:17:08,176 DEBUG i.n.h.c.h.Http2ConnectionHandler: [id: 0x3d46fc58, L:/192.168.0.224:49389 ! R:mrc.server.address/1.1.1.1:443] Sending GOAWAY failed: lastStreamId '0', errorCode '2', debugData ''. Forcing shutdown of the connection.
javax.net.ssl.SSLException: SSLEngine closed already`

In the result it shows:

`Exception: io.grpc.StatusRuntimeException: UNAVAILABLE: ssl exception. []`

The same requests with the same certificates works good in GRPC GUI tool.
I added certificates in pkcs12 format to Jmeter keystore in system.properties file. Request to the same server with HTTP work well - server accepts the certificate.
